operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -146,9 +151,213 @@ private void addInitContainer(Deployment baseDeployment, List<String> extensions
         baseDeployment.getSpec().getTemplate().getSpec().setInitContainers(Collections.singletonList(initContainer));
     }
 
+    public void validatePodTemplate(KeycloakStatusBuilder status) {
+        if (keycloakCR.getSpec() == null ||
+                keycloakCR.getSpec().getUnsupported() == null ||
+                keycloakCR.getSpec().getUnsupported().getPodTemplate() == null) {
+            return;
+        }
+        var overlayTemplate = this.keycloakCR.getSpec().getUnsupported().getPodTemplate();
+
+        if (overlayTemplate.getMetadata() != null &&
+            overlayTemplate.getMetadata().getName() != null) {
+            status.addWarningMessage("The name of the podTemplate cannot be modified");
+        }
+
+        if (overlayTemplate.getMetadata() != null &&
+            overlayTemplate.getMetadata().getNamespace() != null) {
+            status.addWarningMessage("The namespace of the podTemplate cannot be modified");
+        }
+
+        if (overlayTemplate.getSpec() != null &&
+            overlayTemplate.getSpec().getContainers() != null &&
+            overlayTemplate.getSpec().getContainers().get(0) != null &&
+            overlayTemplate.getSpec().getContainers().get(0).getName() != null) {
+            status.addWarningMessage("The name of the keycloak container cannot be modified");
+        }
+
+        if (overlayTemplate.getSpec() != null &&
+            overlayTemplate.getSpec().getContainers() != null &&
+            overlayTemplate.getSpec().getContainers().get(0) != null &&
+            overlayTemplate.getSpec().getContainers().get(0).getImage() != null) {
+            status.addWarningMessage("The image of the keycloak container cannot be modified using podTemplate");
+        }
+    }
+
+    private <T, V> void mergeMaps(Map<T, V> map1, Map<T, V> map2, Consumer<Map<T, V>> consumer) {
+        var map = new HashMap<T, V>();
+        Optional.ofNullable(map1).ifPresent(e -> map.putAll(e));
+        Optional.ofNullable(map2).ifPresent(e -> map.putAll(e));
+        consumer.accept(map);
+    }
+
+    private <T> void mergeLists(List<T> list1, List<T> list2, Consumer<List<T>> consumer) {
+        var list = new ArrayList<T>();
+        Optional.ofNullable(list1).ifPresent(e -> list.addAll(e));
+        Optional.ofNullable(list2).ifPresent(e -> list.addAll(e));
+        consumer.accept(list);
+    }
+
+    private <T> void mergeField(T value, Consumer<T> consumer) {
+        Optional.ofNullable(value).ifPresent(v -> {
+            if (v != null && (!(v instanceof List) || (v instanceof List && ((List<?>) v).size() > 0))) {
+                consumer.accept(v);
+            }
+        });
```

Comment:
BTW, just noticed this could be also simplified as the if we have there is enough, no need for `Optional`. We can even omit some redundant conditions.
```suggestion
        if (value != null && (!(value instanceof List) || ((List<?>) value).size() > 0)) {
            consumer.accept(value);
        }
```

Reply:
fixed